### ISSUE TITLE
Question: why is .flexibleHeight a default for non-scrolling UIComponentViewControllers?

### ISSUE THREAD

**Issue by AndrewLipscomb** [alexdrone/Render#107]
Did an upgrade to Neutrino - its looking great ;) Syntax is much nicer and we are seeing some great improvements in speed with the more granular animations. 

Question though - once we finished the upgrade I noticed that all of our views were very vertically stretched. We found that this was due to the default option for UIComponentCanvasOption being for .flexibleHeight, setting the canvas height to the maximum.

Our views currently consist of a common header-content-footer layout, which means knowing the view height is important for figuring out the content window size (with flex). Our popup view controllers' views also need to be aware of their height as they do not scroll. 

In short - is that a sensible default for a non-scrolling view controller? Especially given the UIScrollableComponentViewController class is available?

**Comment by alexdrone** [comment]
I think I don't fully understand the question sorry :(

**Comment by AndrewLipscomb** [comment]
The question is more around what is the "default" way of doing views inside of iOS (I am C++/JS programmer generally, so my knowledge of what is expected may be very wrong ;) ) 

We forked and made a little tweak for this [here](https://github.com/ALTinners/Render/commit/dbbec68ab16e44626302b2afe25583f30b4eacf2) - this means that when we reference layoutSpec.canvasSize, the default height we see from that is the available height on screen for that canvas, as opposed to the ~32000 max set [here](https://github.com/alexdrone/Render/blob/cdb3bb3d1fbb283161f2d673eac63d8b0319c6e0/render/src/Component.swift#L177). If we want to see a large height like that, we explicitly specify it. 

My big question is whether that is by design or for a normal use case we personally don't use - as my knowledge of iOS is somewhat limited I won't assume that we are doing what everyone else expects by default.
